staging/src/k8s.io/legacy-cloud-providers/gce/gce_instancegroup.go

```diff
@@ -60,6 +62,21 @@ func (g *Cloud) ListInstanceGroups(zone string) ([]*compute.InstanceGroup, error
 	return v, mc.Observe(err)
 }
 
+// ListInstanceGroupsWithPrefix lists all InstanceGroups in the project and
+// zone with given prefix.
+func (g *Cloud) ListInstanceGroupsWithPrefix(zone string, prefix string) ([]*compute.InstanceGroup, error) {
+	ctx, cancel := cloud.ContextWithCallTimeout()
+	defer cancel()
+
+	mc := newInstanceGroupMetricContext("list", zone)
+	f := filter.None
+	if prefix != "" {
```

Comment:
hmm, technically currently this will return all instance groups when the prefix specified is empty, which i think should be fine as there is a logically correct response for that input. 

If you think this function should error on empty prefix I can update the comment to include that in the api?

Reply:
Maybe just add a comment saying this function is called only with non-empty prefix. the behavior to list all when prefix is empty is correct, but it feels like an unnecessary step given the current usecase. Adding the comment is probably enough for now.